packages/nextjs/pages/debug.tsx

```diff
@@ -1,17 +1,14 @@
-import { useEffect, useState } from "react";
+import { useState } from "react";
+import contracts from "../generated/hardhat_contracts";
 import type { NextPage } from "next";
 import { ContractUI } from "~~/components/scaffold-eth";
-import { useDeployedContractNames } from "~~/hooks/scaffold-eth";
+import { ContractName } from "~~/hooks/scaffold-eth/contract.types";
+import scaffoldConfig from "~~/scaffold.config";
 
 const Debug: NextPage = () => {
-  const contractNames = useDeployedContractNames();
-  const [selectedContract, setSelectedContract] = useState<string>();
-
-  useEffect(() => {
-    if (!selectedContract && contractNames.length) {
-      setSelectedContract(contractNames[0]);
-    }
-  }, [contractNames, selectedContract]);
+  const contractsData = contracts[scaffoldConfig.targetNetwork.id as unknown as keyof typeof contracts]?.[0]?.contracts;
+  const contractNames = contractsData ? (Object.keys(contractsData) as ContractName[]) : [];
```

Comment:
(Nitpick alert :D)

This is a bit rough to read. I see 3 options:

1. Just a comment: like `// Gets all the deployed contract in the target network....`
2. Have a util (above `Debug` not exported): like `getDeployedContractsInTargetNetwork()`
3. Same as 2, but in a util file, exported.

I'd personally go with 1 or 2.

Reply:
Yeah, agreed. I'll move it to a util file and clean it up a bit :+1: